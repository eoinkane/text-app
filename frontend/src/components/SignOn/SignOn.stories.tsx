import React from "react";
import SignOnComponent from "./SignOn";

export default {
  title: "SignOn",
  component: SignOnComponent,
};

export const SignOn = () => {
  return <SignOnComponent />;
};
