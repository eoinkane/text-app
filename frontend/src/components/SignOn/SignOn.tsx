import React, { useRef } from 'react';
import makeStyles from '@material-ui/core/styles/makeStyles';
import TextBox from '../TextBox';

interface SignOnBaseProps {}

interface SignOnLoginProps extends SignOnBaseProps {
  login: true;
}

interface SignOnSignUpProps extends SignOnBaseProps {
  signup: true;
}

type SignOnProps = SignOnLoginProps | SignOnSignUpProps;

const useStyles = makeStyles({
  container: {
    backgroundColor: "#f5f5f5",
    maxWidth: "inherit",
  },
})

const SignOn: React.FC<SignOnProps> = ({children, ...mode}) => {
  const box1 = useRef<HTMLInputElement>(null);
  const classes = useStyles();

  const r = (() => {
    if (mode.hasOwnProperty("login")) return ""
    return undefined
  })();

  return (
    <>
      <TextBox id="Name" user product="Name" ref={box1}></TextBox>
    </>
  );
};

export default SignOn;
