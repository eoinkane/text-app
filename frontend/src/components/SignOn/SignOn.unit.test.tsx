import React from "react";
import { shallow } from "enzyme";
import SignOn from "./index";

describe("SignOn", () => {
  it("Renders ", () => {
    const wrapper = shallow(<SignOn />);
    expect(wrapper).toMatchSnapshot();
  });
});
