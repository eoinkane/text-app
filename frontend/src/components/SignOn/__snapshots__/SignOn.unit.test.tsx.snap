// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignOn Renders  1`] = `
<Fragment>
  <ForwardRef
    id="Name"
    product="Name"
    user={true}
  />
</Fragment>
`;
